packages/swap/src/helpers/orders/getBestBuyPoll.ts

```diff
@@ -0,0 +1,42 @@
+import {Balance, Swap} from '@yoroi/types'
+
+import {Quantities} from '../../utils/quantities'
+import {getBuyAmount} from './getBuyAmount'
+
+/**
+ * Find the best pool to buy based on the desired sell amount in a liquidity pool.
+ *
+ * @param pools - The liquidity pool list.
+ * @param sell - The desired sell amount.
+ *
+ * @returns The best pool to sell
+ * if the balance in the pool is insuficient it wont throw an error
+ * if the pools balance is 0 it will return undefined
+ * if the pool list is empty it will return undefined
+ */
+export const getBestBuyPool = (
```

Comment:
More of a personal preference but I find a reducer more readable

```ts
type Accumulator = {quantity: BalanceQuantity; pool?: Swap.Pool}
export const getBestBuyPool = (
  pools: Swap.Pool[],
  sell: Balance.Amount,
): Swap.Pool | undefined => {
  const result = pools.reduce(
    (best, pool): Accumulator => {
      const {quantity} = getBuyAmount(pool, sell)
      if (BigInt(quantity) > BigInt(best.quantity)) return {quantity, pool}
      return best
    },
    {quantity: Quantities.zero, pool: undefined} as Accumulator,
  )
  return result.pool
}
```

Haven't tested it

Reply:
It's looks good. but in this case I prefer to use the old style code, it easier to support for me.